src/scripts/migrate_utils.py

```diff
@@ -0,0 +1,257 @@
+
+        Side effects: Also sets self.direct_parent_curies"""
+        # Get params from wherever supplied
+        prefix_map = owned_prefix_map if owned_prefix_map else self.owned_prefix_map
+        onto = ontology if ontology else self.ontology
+        if not (prefix_map and onto):
+            raise ValueError('Term._get_direct_owned_parents(): Must supply both `ontology` and `owned_prefix_map`.')
+
+        self.direct_parent_uris = _get_direct_owned_parent_uris(
+            uri=self.uri, ontology=self.ontology, owned_prefix_map=self.owned_prefix_map)
+        if self.direct_parent_uris:
+            self.direct_parent_curies = [self._converter.compress(x) for x in self.direct_parent_uris]
+        return self.direct_parent_uris
+
+    def fetch_missing_props(
+        self, ontology: ProntoImplementation = None, general_prefix_map: Dict[PREFIX, URI] = None,
+        owned_prefix_map: Dict[PREFIX, URI] = None
+    ):
+        """For any undefined properties, try to set their values using available params.
+
+        ontology: Requires use of `oaklib`, particularly the `ProntoImplementation`.
+        owned_prefix_map: A prefix_map that includes *only* of the prefixes 'owned' by the ontology. Used to
+        general_prefix_map: A general prefix_map. For conversion of URIs and CURIEs, even if not owned by ontology."""
+        # Get params from wherever supplied
+        ontology = ontology if ontology else self.ontology
+        general_prefix_map = general_prefix_map if general_prefix_map else self.general_prefix_map
+        owned_prefix_map = owned_prefix_map if owned_prefix_map else self.owned_prefix_map
+        if not (ontology or general_prefix_map or owned_prefix_map):
+            raise ValueError('Term.fetch_missing_props(): Must supply at least one param.')
+
+        # Set: uri, curie
+        prefix_map = general_prefix_map if general_prefix_map else owned_prefix_map
+        if prefix_map and self.uri and not self.curie:
+            converter = self._converter if self._converter else curies.Converter.from_prefix_map(prefix_map)
+            curie = converter.compress(self.uri)
+            self.curie = curie if curie != self.uri else None
+        elif prefix_map and self.curie and not self.uri:
+            converter = curies.Converter.from_prefix_map(prefix_map)
+            uri = converter.expand(self.curie)
+            self.uri = uri if uri != self.curie else None
+
+        # Set: label, definition
+        # todo: When OAK supports: I feel like this would be faster to do outside the Term class itself
+        #  ...e.g. build a dict of CURIE-> Term, then use use ontology.labels() on all curies and set label
+        #  ...Right now, OAK ProntoImplementation labels() is just proxy for label(), though SqlImplementation has, and
+        #  ...Both of them have definition() but not definitions().
+        if ontology:
+            self.label = ontology.label(self.curie)
```

Comment:
### Label issues
@matentzn Only having success getting labels for: DOID, NCIT. Getting 100% of them. FOR OMIM, ORDO, ICD10CM, and ICD10WHO, getting 0% of labels.

I created an issue in OAK: https://github.com/INCATools/ontology-access-kit/issues/276

### Definition issues
The line of code which this comment is on is where I'm getting the label. The very next line is where I'm getting the definition.

I'm only getting definitions for DOID, and only for about 11% of them. I see that they are defined by `IAO_0000115` on class annotations. From what I could see, none of the other ontologies had `IAO_0000115` present on any annotations. Perhaps this is why no definitions?

---

@matentzn How shall I proceed?
a. **Wait until OAK fixes**, assuming getting the labels and definitions here are not critical for `slurp`.
b. **I can try to contribute to OAK to fix the issue**. I know about where to do the work and have some idea of how to do it, but this would probably take time and hold us up.
c. **SPARQL**. As I mentioned, I can use SPARQL for now. However, if these are not a big deal for `slurp`, I would prefer (a) in the interest of time, or at least finish up the `slurp` and start using it, and then I can immediately start working on a ticket to add labels/definitions via SPARQL.

Reply:
Label issues solved. Not sure why I wasn't getting any. I'm getting them now...

We made an issue to deal w/ the lack of `IAO_0000115 `